crates/marketplace/src/testing/order_test.rs

```diff
@@ -234,180 +334,163 @@ async fn test_builder_order_chain_fork() {
+            senders
+                .da_proposal
+                .broadcast(da_proposal_msg)
+                .await
+                .unwrap();
+            senders
+                .quorum_proposal
+                .broadcast(quorum_proposal_msg)
+                .await
+                .unwrap();
 
-        // get transactions submitted in previous rounds, [] for genesis
-        // and simulate the block built from those
-        let transactions = prev_proposed_transactions.take().unwrap_or_default();
-        let (quorum_proposal, quorum_proposal_msg, da_proposal_msg, builder_state_id) =
-            calc_proposal_msg(
-                NUM_STORAGE_NODES,
-                round,
-                prev_quorum_proposal,
-                transactions.clone(),
-            )
-            .await;
-
-        let transactions_2 = prev_proposed_transactions_2.take().unwrap_or_default();
-        let (quorum_proposal_2, quorum_proposal_msg_2, da_proposal_msg_2, builder_state_id_2) =
-            calc_proposal_msg(
-                NUM_STORAGE_NODES,
-                round,
-                prev_quorum_proposal_2.clone(),
-                transactions_2.clone(),
-            )
-            .await;
-        if transactions_2 != transactions {
-            fork = true;
-            tracing::debug!("Fork Exist.")
-        } else {
-            fork = false;
-            tracing::debug!("No fork.");
-        }
+            builder_state_id
+        };
+
+        // Simulate consensus deciding on the transactions that are included
+        // in the block, branch 2
+        let BuilderStateId {
+            parent_view: parent_view_branch_2,
+            parent_commitment: parent_commitment_branch_2,
+        } = {
+            // get transactions submitted in previous rounds, [] for genesis
+            // and simulate the block built from those
+            let transactions = prev_proposed_transactions_branch_2
+                .clone()
+                .unwrap_or_default();
+            let (quorum_proposal, quorum_proposal_msg, da_proposal_msg, builder_state_id) =
+                calc_proposal_msg(
+                    NUM_STORAGE_NODES,
+                    round,
+                    prev_quorum_proposal_branch_2,
+                    transactions,
+                )
+                .await;
+
+            prev_quorum_proposal_branch_2 = Some(quorum_proposal.clone());
+
+            // send quorum and DA proposals for this round
+            // we also need to send out the message for the fork-ed chain although it's not forked yet
+            // to prevent builders resend the transactions we've already committed
+            senders
+                .da_proposal
+                .broadcast(da_proposal_msg)
+                .await
+                .unwrap();
+            senders
+                .quorum_proposal
+                .broadcast(quorum_proposal_msg)
+                .await
+                .unwrap();
```

Comment:
I was always wondering whether it's okay to use the same `senders` for fork-ed chain, but they should be able to identify their own message by checking out the commitment so I guess it's fine.